org.eclipse.jdt.core/codeassist/org/eclipse/jdt/internal/codeassist/select/SelectionParser.java

```diff
@@ -104,6 +105,8 @@ public class SelectionParser extends AssistParser {
 	protected static final int K_INSIDE_EXPRESSION_SWITCH = SELECTION_PARSER + 12; // whether we are in an expression switch
 	protected static final int K_INSIDE_WHEN = SELECTION_PARSER + 13; // whether we are in the guard
 
+	protected static final int K_INSIDE_FOR_EACH = SELECTION_OR_ASSIST_PARSER + 14; // whether we are in a for each statement
+
```

Comment:
I think this constant needs to be defined only for the SELECTION_PARSER namespace - cf earlier constants in the same file.